java/src/json/ext/Generator.java

```diff
@@ -334,13 +334,13 @@ void generate(final Session session, RubyHash object,
 
                 buffer.append((byte)'{');
                 buffer.append(objectNl);
-                object.visitAll(new RubyHash.Visitor() {
-                    private boolean firstPair = true;
 
+                final boolean[] firstPair = new boolean[]{true};
+                object.visitAll(new RubyHash.Visitor() {
```

Comment:
This is fine for the patch, but it's now constructing two objects instead of one. Making this an inner class with an accessible firstPair field would get it back down to one object.

Reply:
Ah, read it out of the object field, yeah.

Reply:
It turned out to be quite a lot of refactoring to make the field accessible, as the local holding the visitor needs a new named class or interface in order to be able to read a field out of it, so I didn't do this change. I don't know if you want to @headius? Otherwise I'll leave it as is.